src/hooks/twitch.ts

```diff
@@ -0,0 +1,62 @@
+import useSWR from 'swr';
+import { useTwitchContext } from '../components/TwitchContext';
+import { TWITCH_API_ENDPOINT } from '../constants';
+import {
+  TwitchResponses,
+  TwitchUsersResponse,
+  TwitchBroadcasterSubscriptionsResponse,
+  TwitchUsersFollowsResponse,
+} from '../interfaces';
+
+const requestInit: RequestInit = {
+  method: 'GET',
+  mode: 'cors',
+  cache: 'no-store',
+};
+
+const twitchApiFetcher = async <T extends TwitchResponses>(
+  url: string,
+  headers: HeadersInit
+): Promise<T> => {
+  const response = await fetch(url, { ...requestInit, headers });
+  return response.json();
+};
+
+const useTwitchApi = <T extends TwitchResponses>(path: string | null) => {
+  const { accessToken, clientId } = useTwitchContext();
+  const url = path === null ? null : `${TWITCH_API_ENDPOINT}${path}`;
+  const headers: HeadersInit = {
+    'client-id': clientId,
+    Authorization: `Bearer ${accessToken}`,
+  };
+  return useSWR<T>(url, url => twitchApiFetcher(url, headers));
```

Comment:
If url is null will SWR do nothing?

Reply:
That's right.
https://swr.vercel.app/docs/conditional-fetching